python/lsst/alert/packet/schema.py

```diff
@@ -267,7 +267,8 @@ def retrieve_alerts(self, fp):
             Alert records.
         """
         from .io import retrieve_alerts
-        return retrieve_alerts(fp, reader_schema=self)
+        schema, records = retrieve_alerts(fp, reader_schema=self)
+        return schema, list(records)
```

Comment:
Do we want to convert to a list here? This obviously retains the old behaviour, but it's just going to run into the same memory problems as you fixed in `io.py`. I'd be tempted to just pass through the `records` iterator.

Reply:
(Also, I notice the docstring for this method is wrong. Not due to this ticket, but if you felt like fixing it while you have the file open...!)

Reply:
Woof, you're right. Now I'm fairly worried that the tests are incorrect, as I had this in there while trying to verify the laziness around setting the `writer_schema`... hold please!

Reply:
Okay, thankfully this only affects the test code a bit. Whew.